### PR TITLE
Introduce `Rails::Generators::Testing::Assertions#assert_initializer`

### PR DESCRIPTION
**Pull request by stevepolitodesign**
### Motivation / Background

Compliments the existing [initializer][] generator action.

[initializer]: https://api.rubyonrails.org/classes/Rails/Generators/Actions.html#method-i-initializer

### Detail

```rb
assert_initializer "mail_interceptors.rb"
assert_initializer "mail_interceptors.rb", /SandboxEmailInterceptor/
assert_initializer "mail_interceptors.rb" do |initializer|
  assert_match(/SandboxEmailInterceptor/, initializer)
end
```

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: railties/lib/rails/generators/testing/assertions.rb [hunk 1]
```diff
@@ -121,6 +121,26 @@ def assert_field_default_value(attribute_type, value)
...
+        #
+        #   assert_initializer "mail_interceptors.rb"
+        #
+        # You can also give extra arguments. If the argument is a regexp, it will check if the
+        # regular expression matches the given file content. If it's a string, it compares the
+        # file with the given string:
+        #
+        #   assert_initializer "mail_interceptors.rb", /SandboxEmailInterceptor/
+        #
+        # Finally, when a block is given, it yields the file content:
```
Comment (bogdan) [inline railties/lib/rails/generators/testing/assertions.rb:136]: I don't quite see a block argument usage below.

Comment (rafaelfranca) [reply railties/lib/rails/generators/testing/assertions.rb:136]: True. And since there isn't any test for it, we didn't caught. I'm fixing it

Comment (stevepolitodesign) [reply railties/lib/rails/generators/testing/assertions.rb:136]: There could be value in keeping it for cases like this:

```ruby
expcted = <<~RUBY
  Rails.application.configure do
    if Rails.env.staging?
      config.action_mailer.interceptors = %w[SandboxEmailInterceptor]
    end
  end
RUBY

assert_initializer "mail_interceptors.rb" do |initializer|
  assert_equal expcted, initializer
end
```

Comment (rafaelfranca) [reply railties/lib/rails/generators/testing/assertions.rb:136]: I kept it.

**Pull request merged**